src/robot/conf/settings.py

```diff
@@ -92,6 +94,9 @@ def __setitem__(self, name, value):
     def _process_value(self, name, value):
         if name == 'ReRunFailed':
             return gather_failed_tests(value)
+        if name == 'ReRunFailedSuites':
+            # will be calculated at the end
+            return lambda: gather_failed_suites(value, self.critical_tags, self.non_critical_tags)
```

Comment:
Returning a lambda that is later called in `_process_cli_opts` is a bit ugly but works. Alternative option would be letting the normal option setting logic to set `ReRunFailedSuites` and then in `_process_cli_opts` doing something like this:

```python
if self['ReRunFailedSuites']:
    self['SuiteNames'] += gather_failed_suites(self['ReRunFailedSuites'], self.critical_tags, self.non_critical_tags)
```

For consistency reason `ReRunFailed` should be changed to follow the same logic. What do you think?